### PR TITLE
refactor: config path accessors with instantiation cleanup

### PR DESCRIPTION
**Pull request by lkingland**
Cleans up instantiation logic and config/repo path accessors/creators, fixing a bug or two along the way.

- :broom:  Enables separate paths: when repositories path is outside config path
- :broom:  Removes need to use a client to trigger creation of paths by re-enabling static path accessors
- :broom: Cleans up instantiation logic
- :broom: Removes some unused accessors

/kind techdebt

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -178,54 +180,61 @@ func New(options ...Option) *Client {
...
+	// Trigger the creation of the config and repository paths
+	_ = ConfigPath()         // Config is package-global scoped
+	_ = c.RepositoriesPath() // Repositories is Client-specific
+
 	return c
 }
 
-// assertConfigDir makes a best-effort attempt to create the config directory
-// (including required subdirectories).
-func assertConfigDir() {
```
Comment (lance) [inline client.go:192]: Not just renamed - removed!

File: client.go [hunk 2]
```diff
@@ -178,54 +180,61 @@ func New(options ...Option) *Client {
...
-	// available HOME in which to find .`~/.config/func`.
-	// Since it is expected that the code elsewhere never assume the config
-	// directory exists (doing so is a racing condition), it is valid to simply
-	// handle errors at this level.
-	if err := os.MkdirAll(RepositoriesPath(), 0700); err != nil {
-		fmt.Fprintf(os.Stderr, "Error creating '%v': %v", RepositoriesPath(), err)
-	}
-}
+// The default config path is evaluated in the following order, from lowest
+// to highest precidence.
```
Comment (lance) [inline client.go:200]: Nit - I'm surprised the Knative spell checker didn't catch it :)
```suggestion
// to highest precedence.
```

File: repositories.go [hunk 3]
```diff
@@ -31,9 +31,16 @@ type Repositories struct {
...
+	// Enables single-repository mode.  This replaces the default embedded repo
+	// and extended repositories.  This is an important mode for both diskless
+	// (config-less) operation, such as security-restrited environments, and for
+	// running as a library in which case environmental settings should be
+	// ignored in favor of a more functional approach in which only inputs affect
+	// outputs.
 	remote string
 
-	// backreference to the client enabling full api access for the repo manager
+	// backreference to the client enabling this repositorires manage to
```
Comment (lance) [inline repositories.go:42]:
```suggestion
	// backreference to the client enabling this repositorires manager to
```

**Review by lance — COMMENTED** [review]
Looks good. I just have a couple of extremely small spelling nits.

File: client.go (hunk 2 above)
Comment (lkingland) [reply client.go:200]: Thanks!  Suprised as well...  It's usually so frustratingly thorough :)

**Comment by lkingland** [comment]
> Looks good. I just have a couple of extremely small spelling nits.

Thanks for taking a look!

**Review by lance — APPROVED** [review]
/lgtm

**Pull request merged**